### PR TITLE
Miscellaneous Optimizations and Code Cleanup

### PR DESCRIPTION
**Pull request by Clashsoft**
This pull requests bundles a few very small and local optimizations and code cleanups I made. Does not include syntactic cleanup like formatting.

### CODE REVIEW

File: src/org/stringtemplate/v4/Interpreter.java [hunk 1]
```diff
@@ -495,29 +494,31 @@ void super_new(InstanceScope scope, String name, int nargs) {
...
+            // TODO as this method was extracted from the two super_new overloads,
+            //      there was this difference in which method was called,
+            //      hence the addition of the 'embedded' parameter.
+            //      One of the calls is probably wrong.
+            if (embedded) {
+                st = imported.nativeGroup.getEmbeddedInstanceOf(this, scope, name);
+            }
+            else {
+                st = imported.nativeGroup.createStringTemplateInternally(imported);
+            }
```
Comment (Clashsoft) [inline src/org/stringtemplate/v4/Interpreter.java]: This needs some review. I'm not sure whether this difference was intended, perhaps someone changed one method and forgot the other.

File: src/org/stringtemplate/v4/Interpreter.java [hunk 2]
```diff
@@ -1393,7 +1393,7 @@ protected void trace(InstanceScope scope, int ip) {
     protected void printForTrace(StringBuilder tr, InstanceScope scope, Object o) {
         if ( o instanceof ST ) {
             if ( ((ST)o).impl ==null ) tr.append("bad-template()");
-            else tr.append(" "+((ST)o).impl.name+"()");
+            else tr.append(" ").append(((ST) o).impl.name).append("()");
```
Comment (sharwell) [inline src/org/stringtemplate/v4/Interpreter.java]: 📝 This code base typically omits a space after a cast.

Comment (Clashsoft) [reply src/org/stringtemplate/v4/Interpreter.java]: Good catch! Must have happened when applying the fixit suggestion with IntelliJ.

```suggestion
            else tr.append(" ").append(((ST)o).impl.name).append("()");
```

**Comment by parrt** [comment]
Hiya @Clashsoft @sharwell where are we with respect to finishing off ST changes? I'd like to include in next ANTLR release soon.

**Comment by Clashsoft** [comment]
Hi @parrt , #235 should be ready for merging, #237 is awaiting changes from @sharwell , and I'm nearly done with this one. I also started work on a PR centred on deprecation.

**Comment by Clashsoft** [comment]
Update: this PR is now ready to be merged from my side, please review!

**Comment by parrt** [comment]
Hi. looks like you are starting to rewrite most of the code base with all these PRs ;)  Can we just focus on functionality? I'm not able to spend time to review all this.  I also need to get ANTLR out the door.  if I cut a release right now, is ST4 stable?

**Comment by Clashsoft** [comment]
This PR doesn't need to be in the coming release. Most of the changes are small and improve code quality, some might improve performance slightly. But still, I'd really like to get them in at some point. If you want, I can try and split the admittedly large changeset into smaller PRs.

**Pull request closed without merging**